src/Shared/ThrottlingDomainEventBehavior.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Threading.RateLimiting;
+using Microsoft.Extensions.Options;
+
+namespace Workleap.DomainEventPropagation;
+
+internal class ThrottlingDomainEventBehavior(IOptions<EventPropagationThrottlingOptions> options) : ISubscriptionDomainEventBehavior, IDisposable
+{
+    private readonly RateLimiter _rateLimiter = new TokenBucketRateLimiter(new TokenBucketRateLimiterOptions
```

Comment:
I'm curious about the thought process that led to choosing this throttling strategy, do you have some details?